### PR TITLE
Move PreconditionViolationException to base package

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Addresses #1863

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java [hunk 1]
```diff
@@ -102,7 +102,7 @@
 			Path baseDir = closeablePath.getPath();
 			return findClassesForPath(baseDir, basePackageName, classFilter);
 		}
-		catch (PreconditionViolationException ex) {
+		catch (org.junit.platform.commons.PreconditionViolationException ex) {
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java]: ❓ Can we replace this with an import as well?

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ public static void assertPackageNameIsValid(String packageName) {
 	 * @param name the attribute name as a string
 	 * @return an {@code Optional} containing the attribute value; never
 	 * {@code null} but potentially empty
-	 * @throws PreconditionViolationException if the supplied type is
+	 * @throws org.junit.platform.commons.PreconditionViolationException if the supplied type is
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java]: :x: nit: We either need to import the exception from the new place or re-wrap the lines here to have a max of 80 characters.

Comment (sormuras) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java]: Why doesn't `spotlessApply` take care of it?

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java (hunk 1 above)
Comment (sormuras) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java]: Done.

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java (hunk 2 above)
Comment (sormuras) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java]: Anyway, I manually formatted the lines and did _not_ introduce an explicit `import` for a type only needed in Javadoc context.

Comment (marcphilipp) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java]: The Eclipse formatter does not format comments.

**Pull request merged**